src/Nox.Types/Types/AutoNumber/AutoNumber.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Nox.Types;
+
+/// <summary>
+/// Represents a Nox <see cref="AutoNumber"/> type and value object.
+/// </summary>
+public sealed class AutoNumber : ValueObject<uint, AutoNumber>
+{
+    public AutoNumber()
+    {
+        Value = 0;
+    }
+
+    /// <summary>
+    /// Creates a <see cref="AutoNumber"/> object with the initial value of <paramref name="value"/>.
+    /// </summary>
+    /// <param name="value">Value to be parsed.</param>
+    /// <returns>New <see cref="AutoNumber"/> object.</returns>
+    public new static AutoNumber From(uint value) => new() { Value = value };
+
+    /// <summary>
+    /// Gets the current Value incremented by one.
+    /// </summary>
+    /// <returns>The current value incremented by one.</returns>
+    public uint GetNext() => ++Value;
```

Comment:
no logic is needed, this is kind of a marker class. This type only load the value from database nothing else

Reply:
`GetNext` method removed. DatabaseNumber can still be accessed through `Value`.